File: PettyTest2017/Dagger2/MyTest406_3/MyTest306/app/src/main/java/com/ethanco/mytest306/BaseActivity.java
package com.ethanco.mytest306;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by EthanCo on 2016/4/6.
 */
public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initInject(App.getInstance().getBuilder().appModule(new AppModule(this)).build());
    }

    public abstract void initInject(AppComponent component);
}

File: KoltinTestStore/KoltinTest01/app/src/main/java/com/ethanco/koltintest01/BaseActivity.kt
package com.ethanco.koltintest01

import android.support.v7.app.AppCompatActivity

/**
 *
 * @author EthanCo
 * @since 2017/5/24
 */
open class BaseActivity : AppCompatActivity() {
    val app by lazy { application as App }
    val pref by lazy { app.pref }
}